packages/api/tests/lib/writeRecords/test-write-execution.js

```diff
@@ -514,3 +547,27 @@ test('writeExecutionRecordFromApi() saves execution to Dynamo and RDS with same
   t.is(pgRecord.created_at.getTime(), dynamoRecord.createdAt);
   t.is(pgRecord.updated_at.getTime(), dynamoRecord.updatedAt);
 });
+
+test.serial('writeExecutionRecordFromMessage() calls publishExecutionSnsMessage and successfully publishes an SNS message', async (t) => {
+  const {
+    cumulusMessage,
+    executionArn,
+    knex,
+    QueueUrl,
+  } = t.context;
+
+  await writeExecutionRecordFromMessage({ cumulusMessage, knex });
+
+  const { Messages } = await sqs().receiveMessage({ QueueUrl, WaitTimeSeconds: 10 }).promise();
+
+  t.is(Messages.length, 1);
+
+  const snsMessage = JSON.parse(Messages[0].Body);
+  const executionRecord = JSON.parse(snsMessage.Message);
+  const executionApiRecord = generateExecutionApiRecordFromMessage(cumulusMessage, Date.now());
+  const omitFields = ['updatedAt', 'timestamp'];
+
+  t.is(executionRecord.arn, executionArn);
+  t.is(executionRecord.status, cumulusMessage.meta.status);
+  t.deepEqual(omit(executionRecord, omitFields), omit(executionApiRecord, omitFields));
```

Comment:
We want to prove that the record that is sent to SNS represents the record **after** being written to the database, not before, whereas comparing against `executionApiRecord` is comparing to the record before it's written. Instead, we should get the record out of Dynamo that was written and compare that with what was sent to SNS. Then we also don't want to omit any fields, because we want to prove that what went to SNS was exactly what we expected

Reply:
[`bd0112b` (#2422)](https://github.com/nasa/cumulus/pull/2422/commits/bd0112b50cc0fda564ec0be0a84a830659daf1ea)